### PR TITLE
[Refactor] Autoscaling

### PR DESCRIPTION
**Pull request by Aloento**
Move all API methods and related structures to a separate file.
Inline all `Extracts` and use the `extract` package instead.
Remove all Builders
Remove groups_hcs

Merge request and result structs:
NetworkOpts -> ID
DiskOpts -> Disk
SecurityGroupOpts -> SecurityGroup

Merge the same structures:
CreateOpts -> PolicyOpts

### CODE REVIEW

File: openstack/autoscaling/v1/groups_hcs/delete.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package groups_hcs
```
Comment (anton-sidelnikov) [inline openstack/autoscaling/v1/groups_hcs/delete.go]: hcs - what is it?

Comment (Aloento) [reply openstack/autoscaling/v1/groups_hcs/delete.go]: I don't know but this appears to be code from upstream.
https://github.com/opentelekomcloud/gophertelekomcloud/commit/6bb917f8c1d193a373ca168aea177f5e815edd44
I will check if they are still in use or not.

Comment (Aloento) [reply openstack/autoscaling/v1/groups_hcs/delete.go]: This is an old version of the code that was deprecated, so I removed it.

File: openstack/autoscaling/v1/groups/results.go [hunk 2]
```diff
@@ -80,45 +34,7 @@ type Group struct {
...
-}
-
-type LBaaSListener struct {
-	ListenerID   string `json:"listener_id"`
-	PoolID       string `json:"pool_id"`
-	ProtocolPort int    `json:"protocol_port"`
-	Weight       int    `json:"weight"`
-}
-
-type GroupPage struct {
```
Comment (Aloento) [inline openstack/autoscaling/v1/groups/results.go:103]: Does this pager really help? Do we need to keep it?

File: openstack/autoscaling/v1/groups/create.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+type LBaaSListener struct {
+	ListenerID   string `json:"listener_id"`
+	PoolID       string `json:"pool_id" required:"true"`
+	ProtocolPort int    `json:"protocol_port" required:"true"`
+	Weight       int    `json:"weight" required:"true"`
+}
+
+type ID struct {
+	ID string `json:"id" required:"true"`
+}
```
Comment (outcatcher) [inline openstack/autoscaling/v1/groups/create.go:41]: This seems to be commonly used structure, should be placed in `common`.
image

File: openstack/autoscaling/v1/groups/results.go (hunk 2 above)
Comment (outcatcher) [reply openstack/autoscaling/v1/groups/results.go:103]: This is a `SinglePageBase`, pagination without pagination

**Review by outcatcher — DISMISSED** [review]
Please also add descriptions to structure fields, I loved them in previous PRs :grin:

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**